Drop Debian <= 7, Ubuntu <= 14.04, EL < 7 and SLES < 12

ekohl opened the pull request.

tuxmea commented:
Are there SLES12 packages available for SLES12?
it would be great if we could have SLES 12 OpenVox 7 packages (agent and server + db)
There are some orgs which run sles12 with extended support and my idea is to make the transition smooth.

bastelfreak commented on resources/puppetlabs/lein-ezbake/template/global/Makefile.erb:
```diff
@@ -37,13 +37,6 @@ install-<%= EZBake::Config[:project] %>-termini:
 <% end -%>
 <% end -%>
 
-install-rpm-sysv-init: install-rpm-preinst
```
stupid question, why is there `install-rpm-sysv-init` and `install-rpm-systemd`, but for Debian only `install-deb-sysv-init`?

ekohl replied:
I question if this file is even used at all. Given teh last time it was updated was in 2015 I think it isn't.

bastelfreak commented on resources/puppetlabs/lein-ezbake/template/global/ext/redhat/prerm.erb:
```diff
@@ -4,11 +4,4 @@
-    /sbin/chkconfig --del %{name}
-fi
-%endif
+
```
is that empty line here on purpose?

ekohl replied:
No

ekohl commented:
Updated to also drop the unused Makefile and the whole `systemd` variable for Red Hat based RPMs.

nmburgan commented:
@tuxmea I had problems getting a SLES 12 container to work.  But if it's helpful, I can pick that back up and see if I can get it working.

@ekohl I built openvox-server and openvoxdb with a version of ezbake that includes these changes.  I don't have much time to test them today, but anyone feel free to poke at them and see if there are any issues.

https://artifacts.overlookinfratech.com/#openvox-server/99.99.99/
https://artifacts.overlookinfratech.com/#openvoxdb/99.99.99/

ekohl commented:
Not a real test, but I had a look at the differences. A lot of noise because of an update to the `gli` gem and I guess the binstubs aren't deterministic.
```console
$ rpmdiff -i=T openvox-server-*.el10.noarch.rpm
S.5.....    DESCRIPTION
removed     REQUIRES tzdata-java  
removed     PROVIDES openvox-server = 8.8.0-1.el10
added       PROVIDES openvox-server = 99.99.99-1.el10
S.5........ /opt/puppetlabs/puppet/lib/ruby/vendor_gems/bin/puppetserver-ca
S.5........ /opt/puppetlabs/server/apps/puppetserver/bin/puppetserver
S.5........ /opt/puppetlabs/server/apps/puppetserver/ezbake.manifest
S.5........ /opt/puppetlabs/server/apps/puppetserver/puppet-server-release.jar
S.5........ /opt/puppetlabs/server/apps/puppetserver/scripts/install.sh
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/dropsonde
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/facter
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/gli
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/hocon
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/httparty
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/httpclient
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/puppet
S.5........ /opt/puppetlabs/server/data/puppetserver/dropsonde/bin/thor
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/cache/gli-2.22.1.gem
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/cache/gli-2.22.2.gem
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/.gitignore
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/CONTRIBUTING.md
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/Dockerfile.dx
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/Gemfile
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/LICENSE.txt
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/README.rdoc
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/Rakefile
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/bin
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/bin/ci
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/bin/gli
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/bin/rake
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/bin/setup
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/build
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/docker-compose.env
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/dx.sh.lib
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/exec
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/prune
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/setupkit.sh.lib
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/show-help-in-app-container-then-wait.sh
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/start
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/dx/stop
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/exe
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/exe/gli
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/gli.cheat
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/gli.gemspec
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/gli.rdoc
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/app.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/app_support.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/argument.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/command.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/command_finder.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/command_line_option.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/command_line_token.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/command_support.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/compound_command.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/doc.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/arg_name_formatter.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/command_finder.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/command_help_format.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/compact_synopsis_formatter.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/full_synopsis_formatter.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/global_help_format.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/help_completion_format.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/list_formatter.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/one_line_wrapper.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/options_formatter.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/terminal_synopsis_formatter.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/text_wrapper.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/tty_only_wrapper.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/help_modules/verbatim_wrapper.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/initconfig.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/rdoc_document_listener.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/commands/scaffold.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/dsl.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/exceptions.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/flag.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/gli_option_block_parser.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/gli_option_parser.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/option_parser_factory.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/option_parsing_result.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/options.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/switch.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/terminal.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/lib/gli/version.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/object-model.dot
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/object-model.png
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/README.md
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/Gemfile
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/README.rdoc
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/Rakefile
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/bin
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/bin/todo
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo/commands
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo/commands/create.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo/commands/list.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo/commands/ls.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo/commands/make.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/lib/todo/version.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/test
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/test/tc_nothing.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/todo.gemspec
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo/todo.rdoc
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/Gemfile
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/README.rdoc
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/Rakefile
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/bin
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/bin/todo
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib/todo
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib/todo/commands
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib/todo/commands/create.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib/todo/commands/list.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib/todo/commands/ls.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/lib/todo/version.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/test
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/test/tc_nothing.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/todo.gemspec
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_legacy/todo.rdoc
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_plugins
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_plugins/commands
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/apps/todo_plugins/commands/third.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/integration
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/integration/gli_cli_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/integration/gli_powered_app_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/integration/scaffold_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/integration/test_helper.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/command_finder_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/command_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/compound_command_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/doc_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/flag_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/gli_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/help_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/init_simplecov.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/options_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/subcommand_parsing_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/subcommands_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/support
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/support/fake_std_out.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/support/gli_test_config.yml
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/switch_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/terminal_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/test_helper.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.1/test/unit/verbatim_wrapper_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/.gitignore
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/CONTRIBUTING.md
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/Dockerfile.dx
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/Gemfile
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/LICENSE.txt
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/README.rdoc
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/Rakefile
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/bin
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/bin/ci
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/bin/gli
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/bin/rake
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/bin/setup
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/build
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/docker-compose.env
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/dx.sh.lib
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/exec
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/prune
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/setupkit.sh.lib
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/show-help-in-app-container-then-wait.sh
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/start
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/dx/stop
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/exe
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/exe/gli
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/gli.cheat
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/gli.gemspec
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/gli.rdoc
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/app.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/app_support.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/argument.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/command.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/command_finder.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/command_line_option.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/command_line_token.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/command_support.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/compound_command.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/doc.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/arg_name_formatter.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/command_finder.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/command_help_format.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/compact_synopsis_formatter.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/full_synopsis_formatter.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/global_help_format.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/help_completion_format.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/list_formatter.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/one_line_wrapper.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/options_formatter.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/terminal_synopsis_formatter.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/text_wrapper.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/tty_only_wrapper.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/help_modules/verbatim_wrapper.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/initconfig.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/rdoc_document_listener.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/commands/scaffold.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/dsl.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/exceptions.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/flag.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/gli_option_block_parser.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/gli_option_parser.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/option_parser_factory.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/option_parsing_result.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/options.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/switch.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/terminal.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/lib/gli/version.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/object-model.dot
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/object-model.png
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/README.md
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/Gemfile
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/README.rdoc
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/Rakefile
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/bin
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/bin/todo
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo/commands
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo/commands/create.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo/commands/list.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo/commands/ls.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo/commands/make.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/lib/todo/version.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/test
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/test/tc_nothing.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/todo.gemspec
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo/todo.rdoc
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/Gemfile
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/README.rdoc
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/Rakefile
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/bin
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/bin/todo
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib/todo
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib/todo/commands
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib/todo/commands/create.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib/todo/commands/list.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib/todo/commands/ls.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/lib/todo/version.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/test
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/test/tc_nothing.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/todo.gemspec
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_legacy/todo.rdoc
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_plugins
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_plugins/commands
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/apps/todo_plugins/commands/third.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/integration
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/integration/gli_cli_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/integration/gli_powered_app_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/integration/scaffold_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/integration/test_helper.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/command_finder_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/command_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/compound_command_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/doc_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/flag_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/gli_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/help_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/init_simplecov.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/options_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/subcommand_parsing_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/subcommands_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/support
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/support/fake_std_out.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/support/gli_test_config.yml
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/switch_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/terminal_test.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/test_helper.rb
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/gems/gli-2.22.2/test/unit/verbatim_wrapper_test.rb
removed     /opt/puppetlabs/server/data/puppetserver/dropsonde/specifications/gli-2.22.1.gemspec
added       /opt/puppetlabs/server/data/puppetserver/dropsonde/specifications/gli-2.22.2.gemspec
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/hocon
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/rake
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/rmsgcat
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/rmsgfmt
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/rmsginit
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/rmsgmerge
S.5........ /opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems/bin/rxgettext
S.5........ /usr/lib/systemd/system/puppetserver.service
```

tuxmea commented:
> @tuxmea I had problems getting a SLES 12 container to work. But if it's helpful, I can pick that back up and see if I can get it working.
> 
I have contacted SuSE. Let's see if we are allowed to use their SLES containers.

nmburgan commented:
@tuxmea I'm currently using their container for SLES 15
https://github.com/OpenVoxProject/vanagon/blob/38e78d96bd14510e236a1f5137ad87e67a110dfe/lib/vanagon/platform/defaults/sles-15-x86_64.rb#L24

They have one for 12, but I ran into issues with the openvox build on it.  I don't recall what they were, but I should probably try again, since it looks like they updated the image since I last tried.

@ekohl Looks right, just need to make sure the service still works correctly, I think.  I'd feel more confident if we had any real automated testing :(

ekohl commented:
> @ekohl Looks right, just need to make sure the service still works correctly, I think. I'd feel more confident if we had any real automated testing :(

Agreed on the automated testing.

I'd really like to move away from FPM and use native packaging mechanisms where possible. If you use that then at least for RPMs you could use [packit](https://packit.dev) to create RPMs for every PR to `puppetserver` and run end-to-end tests with it using [Testing Farm](https://docs.testing-farm.io).

This cleanup is IMHO a good start to see what actually needs to be done. Some time ago I also started the other way around: use the tarballs from https://downloads.puppetlabs.com/puppet/index.html and package that up. Results are in https://copr.fedorainfracloud.org/coprs/ekohl/puppetserver. That's slightly different because it attempts to use the system Ruby (not great with JRuby) but it's where https://github.com/OpenVoxProject/ezbake/pull/5 came from.

Taking a step back: what do you think would be reasonable minimal testing? Start the service, configure an agent and see if you can apply a manifest that changes something?

nmburgan commented:
> Taking a step back: what do you think would be reasonable minimal testing? Start the service, configure an agent and see if you can apply a manifest that changes something?

Yep, I think that'd be good.  Testing the openvoxdb package too would be nice, but if the changes work for openvox-server, I don't know why they'd screw up openvoxdb.

ekohl commented:
Trivial rebase since there are more tests now.
